### PR TITLE
[REVIEW] Fix issue related to reading a nullable boolean column in `read_parquet` when `engine=pyarrow`

### PR DESCRIPTION
**Pull request by galipremsagar**
Fixes: #6356 

This PR fixes issue where pyarrow table schema has `boolean` instead of `bool` in `np_dtypes`. With this fix we are instead relying to `pandas_dtypes` when there is a `bool`.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by galipremsagar** [comment]
rerun tests

File: python/cudf/cudf/tests/test_parquet.py [hunk 1]
```diff
@@ -1286,3 +1286,22 @@ def test_parquet_writer_sliced(tmpdir):
...
+        {
+            "a": pd.Series(
+                [True, False, None, True, False], dtype=pd.BooleanDtype()
+            )
+        }
+    )
+    expected_gdf = cudf.DataFrame({"a": [True, False, None, True, False]})
+
+    pdf.to_parquet(pandas_path)
+    actaul_gdf = cudf.read_parquet(pandas_path, engine=engine)
```
Comment (brandon-b-miller) [inline python/cudf/cudf/tests/test_parquet.py:1305]:
```suggestion
    actual_gdf = cudf.read_parquet(pandas_path, engine=engine)
```

File: python/cudf/cudf/tests/test_parquet.py [hunk 2]
```diff
@@ -1286,3 +1286,22 @@ def test_parquet_writer_sliced(tmpdir):
...
+
+    assert_eq(actaul_gdf, expected_gdf)
```
Comment (brandon-b-miller) [inline python/cudf/cudf/tests/test_parquet.py:1307]:
```suggestion
    assert_eq(actual_gdf, expected_gdf)
```

**Review by brandon-b-miller — CHANGES_REQUESTED** [review]
Very minor misspelling otherwise LGTM.

As an aside, do we have this same issue for the other pandas nullable types, eg `string`, `Int64`, etc?

**Comment by galipremsagar** [comment]
> As an aside, do we have this same issue for the other pandas nullable types, eg `string`, `Int64`, etc?

Nope, this is only with `boolean` nullable types. Rest of the types work well.

**Review by isVoid — APPROVED** [review]

**Comment by galipremsagar** [comment]
rerun tests

**Review by brandon-b-miller — APPROVED** [review]

**Comment by galipremsagar** [comment]
rerun tests

**Pull request merged**